publish.js

```diff
@@ -258,22 +258,41 @@ publish.tagVC = function (isTest, version, options) {
 };
 
 /**
- * Restore the package.json file to the latest committed version.
+ * Restore the files specified in filesToClean option to the latest committed version.
+ * Restores package.json and package-lock.json files to the latest commited version by default.
  *
- * Used internally to reset version number changes in package.json
- * @param moduleRoot {String} - the directory where the package.json file to
+ * Used internally to reset version number changes in list of files in filesToClean option.
+ * First checks whether the list of files in filesToClean option are being tracked by git or not.
+ * @param moduleRoot {String} - the directory where the files to 
                                 clean is located in.
- * @param options {Object} - e.g. {"cleanCmd": "git checkout -- package.json"}
+ * @param options {Object} - e.g. {"checkFilesTracking": "git ls-files --error-unmatch ${filesToClean}", "cleanCmd": "git checkout -- ${filesToClean}", "filesToClean": "package.json package-lock.json"}
  */
 publish.clean = function (moduleRoot, options) {
     var originalDir = process.cwd();
 
     // change to the module root directory
     process.chdir(moduleRoot || "./");
 
-    // run the clean command
-    publish.execSyncFromTemplate(options.cleanCmd);
+    var file = options.filesToClean.split(" ");
+    
+    file.forEach(function(file){
+        var isTracked = true;
 
+        // check whether file is being tracked or not
+        try {publish.execSyncFromTemplate(options.checkFilesTracking, {
+                filesToClean: file
+            }, options.checkFilesTrackingHint);
+        } catch (error) {
```

Comment:
is there a way to check for a specific error, just in case something else goes wrong, we may not want to treat all errors as the file not being tracked.

Reply:
Hi, @jobara. It is that git returns the full name of the file if it is present and is being tracked. Can I use this as a parameter to differentiate between errors?

Reply:
hmm.. i'm not sure that will help distinguish the errors, because if an occurs for any reason, you may not get the file name back.

Reply:
@jobara I am now able to extract return codes for errors that I think is a better approach. Here, git returns 0 if command was run successfully and 1 if the file hasn't been added or is being tracked. What should be the expected behaviour for error codes other than 0 and 1?

Reply:
That would be the exit code, which might be useful depending on what other exit codes are produced. However, it's possible that there may be multiple reasons for exiting with the same code when in an error state. Is there any useful information provided in the error object that you get in the catch?

Reply:
@jobara Error object also returns an error message in form of: 
```
Command failed: git ls-files --error-unmatch package-lock.json
error: pathspec 'package-lock.json' did not match any file(s) known to git.
Did you forget to 'git add'?
```
How do you suggest using this information?

Reply:
It doesn't sound like we get a lot of detail on the exact error except through the message returned. I think we will want to log a message that indicates that we suspect that that the file is not tracked or not present and provide the error message that was provided. This way if something else is failing at least the user will know what happened.